### ISSUE TITLE
Serialised POCO Schema Validation

### ISSUE THREAD

**Issue by ethanae** [Elfocrash/Cosmonaut#22]
Hi,

Nice work on the library! Regarding setting up a schema and validating of documents off of that schema. Does something like this already exist in the library? If not, is it on the map?

This library is something C# and Cosmos need. So I'd like to help out.

Regards,
Ethan

**Comment by Elfocrash** [comment]
Hello and thanks. 

Are you talking about, validating the schema of the document being inserted in a collection?

**Comment by ethanae** [comment]
Yes, exactly. Something similar to Mongoose schemas.

**Comment by Elfocrash** [comment]
After taking a look at Mongoose and how Schemas work there here is what I think.

* Cosmonaut was built from the ground up with the idea that the POCOs drive the collection schema. If you change the POCO's properties then it will either not be returned (if you removed it) or it will default to the property's default value (if you add it).
* One CosmosStore is limited to a single POCO (because it is part of the method via the generic type `TEntity`) so it is limited to one "schema".
* MongoDB and CosmosDB work very differently mainly because of pricing. One of the best things Comsonaut has to offer is Collection Sharing. If I add schema restrictions on top of that it will become too messy.
* There is always the danger of someone adding a random object in the collection externally and how the library should handle this.

Basically I don't want an ORM to have too much responsibility.

**Comment by ethanae** [comment]
I see where you are coming. Thanks for the info!
